apple/swift-docc-render
# Calculate image sizes when not provided by Render JSON

Bug/issue #, if applicable: 88195563

## Summary

This adds logic to calculate the optimal width for rendering an image when the size information has not been provided in the Render JSON.

Since DocC does not provide image size information, there may be a situation where a user only provides a 2x version of an image—on 1x displays, this image may end up looking twice as big as was intended. By calculating the appropriate width in the renderer, it can be shown at the expected size regardless of the pixel density of the display.

## Testing

Steps:
1. Download/unzip [FakePackage.doccarchive.zip](https://github.com/apple/swift-docc-render/files/8674002/FakePackage.doccarchive.zip)
2. Run the dev server with `env VUE_APP_DEV_SERVER_PROXY=/path/to/unzipped/FakePackage.doccarchive npm run serve`
3. Open http://localhost:8080/documentation/fakepackage
4. Using real displays and/or responsive design modes, verify that the image is always rendered at 50 pixels wide, regardless of whether or not the display/tool is in 1x or 2x mode
5. Test other existing content in realistic scenarios if possible and verify that there are no regressions with sizing of images when the Render JSON already includes size information

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
